Move data dir to SD crashes the app

- Go to Settings - System
- Select to move the geocache data to the external SD

- Crash

This was reported for the market version 2019.04.16 on Android 4.4.2 but seems also to happen on my Android 9 device

Fix for this should target releas branch.

Not sue if this is an unfixed Targetsdk regression or something else.
Need to take a deeper look next days, unless someone is faster.

It's some layout inflating problem. LogCat is complaining:

```
   android.view.InflateException: Binary XML file line #26: Binary XML file line #44: Error inflating class <unknown>
    Caused by: android.view.InflateException: Binary XML file line #44: Error inflating class <unknown>
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
        at android.view.LayoutInflater.createView(LayoutInflater.java:645)
[...]
        at android.app.AlertDialog.onCreate(AlertDialog.java:423)
        at android.app.ProgressDialog.onCreate(ProgressDialog.java:198)
        at android.app.Dialog.dispatchOnCreate(Dialog.java:389)
        at android.app.Dialog.show(Dialog.java:293)
        at cgeo.geocaching.activity.Progress.show(Progress.java:55)
        at cgeo.geocaching.storage.LocalStorage.changeExternalPrivateCgeoDir(LocalStorage.java:336)
[...]
   Caused by: java.lang.UnsupportedOperationException: Failed to resolve attribute at index 3: TypedValue{t=0x2/d=0x1010036 a=1}
        at android.content.res.TypedArray.getColorStateList(TypedArray.java:528)
        at android.widget.TextView.<init>(TextView.java:775)
        at android.widget.TextView.<init>(TextView.java:704)
        at android.widget.TextView.<init>(TextView.java:700)
        at com.android.internal.widget.DialogTitle.<init>(DialogTitle.java:41)
        at java.lang.reflect.Constructor.newInstance0(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:430) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:645) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:858) 
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:861) 
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821) 
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:994) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:854) 
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:518) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:426) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:377) 
        at com.android.internal.policy.PhoneWindow.setContentView(PhoneWindow.java:414) 
        at com.android.internal.app.AlertController.installContent(AlertController.java:236) 
        at android.app.AlertDialog.onCreate(AlertDialog.java:423) 
        at android.app.ProgressDialog.onCreate(ProgressDialog.java:198) 
        at android.app.Dialog.dispatchOnCreate(Dialog.java:389) 
        at android.app.Dialog.show(Dialog.java:293) 
        at cgeo.geocaching.activity.Progress.show(Progress.java:55) 
        at cgeo.geocaching.storage.LocalStorage.changeExternalPrivateCgeoDir(LocalStorage.java:336) 
        at cgeo.geocaching.settings.SettingsActivity.lambda$null$4$SettingsActivity(SettingsActivity.java:333) 
        at cgeo.geocaching.settings.-$$Lambda$SettingsActivity$xe5YyUxpWGDqicGAfhf2okVoqDw.onClick(lambda) 
        at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:164) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6077) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756) 
```

I was able to partially fix it by adding specific values to `?text_color` parameters of `cgeoProgressdialogTheme` and `cgeoProgressdialogTheme_light` in `styles.xml`, but still get error messages, this time happening at

```
    android.view.InflateException: Binary XML file line #80: Binary XML file line #37: Error inflating class <unknown>
    Caused by: android.view.InflateException: Binary XML file line #37: Error inflating class <unknown>
```

Any idea on how to find out which attribute is missing?

Last time I've dug into this I stopped at some layout inflation problem I could not resolve.

Anybody can tell me how to find out which resource is missing when I get the following error:

```
android.view.InflateException: Binary XML file line #80: Binary XML file line #37: Error inflating class <unknown>
    Caused by: android.view.InflateException: Binary XML file line #37: Error inflating class <unknown>
```

Which "binary XML file" is referenced? Can I un-binary it? Any other way to find out which resource / attribute etc. is missing?

--- a/main/res/values-v1/styles.xml
+++ b/main/res/values-v1/styles.xml
@@ -1,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <!-- progress dialog theme -->
+    <style name="cgeoProgressdialogTheme" parent="cgeoProgressdialogTheme_Base">
+        <item name="android:textColorPrimary">?text_color</item>
+        <item name="buttonBarButtonStyle">@style/button_borderless</item>
+    </style>
+    <style name="cgeoProgressdialogTheme_light" parent="cgeoProgressdialogTheme_light_Base">
+        <item name="android:textColorPrimary">?text_color</item>
+        <item name="buttonBarButtonStyle">@style/button_borderless</item>
+    </style>
+
+</resources>

--- a/main/res/values-v21/styles.xml
+++ b/main/res/values-v21/styles.xml
@@ -1,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <!-- progress dialog theme -->
+    <style name="cgeoProgressdialogTheme" parent="cgeoProgressdialogTheme_Base">
+    </style>
+    <style name="cgeoProgressdialogTheme_light" parent="cgeoProgressdialogTheme_light_Base">
+    </style>
+
+</resources>

--- a/main/res/values/styles.xml
+++ b/main/res/values/styles.xml
@@ -377,17 +377,13 @@
     </style>
 
     <!-- progress dialog theme -->
-    <style name="cgeoProgressdialogTheme" parent="Theme.AppCompat.Dialog.Alert">
-        <item name="android:textColorPrimary">?text_color</item>
+    <style name="cgeoProgressdialogTheme_Base" parent="Theme.AppCompat.Dialog.Alert">
         <item name="android:textColor">?text_color</item>
         <item name="android:windowBackground">@drawable/border_straight</item>
-        <item name="buttonBarButtonStyle">@style/button_borderless</item>
     </style>
-    <style name="cgeoProgressdialogTheme_light" parent="Theme.AppCompat.Light.Dialog.Alert">
-        <item name="android:textColorPrimary">?text_color</item>
+    <style name="cgeoProgressdialogTheme_light_Base" parent="Theme.AppCompat.Light.Dialog.Alert">
         <item name="android:textColor">?text_color</item>
         <item name="android:windowBackground">@drawable/border_straight_light</item>
-        <item name="buttonBarButtonStyle">@style/button_borderless</item>
     </style>
 
     <!-- map progressbar style -->

